packages/protoplugin-test/src/custom-options.test.ts

```diff
@@ -0,0 +1,298 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getDescriptorSet } from "./helpers.js";
+import type { DescFile } from "@bufbuild/protobuf";
+import {
+  getCustomOptionBoolean,
+  getCustomOptionFloat,
+  getCustomOptionUint32,
+  getCustomOptionInt32,
+  getCustomOptionInt64,
+  getCustomOptionString,
+  getCustomOptionMessage,
+  getCustomOptionEnum,
+} from "@bufbuild/protoplugin/ecmascript";
+import { proto3 } from "@bufbuild/protobuf";
+
+// Used for custom options of type FooMessage
+const foo = proto3.makeMessageType("FooMessage", [
```

Comment:
As the author of a plugin with custom options, you provide two things:
1. the code generator plugin
2. the proto file defining the custom options

Since you are in control, you can simply generate the code for the proto file, and use the generated `class FooMessage` instead of `makeMessageType`. That should be a lot easier to use for plugin authors?

Sorry if I lead you on the wrong path with `makeMessageType`. The idea was to get the field information from the schema, which is something we have to do anyways if we want to expose custom options in the generated code. We can get by without it.